### PR TITLE
chore(deps): move from @hapi/joi to joi

### PR DESCRIPTION
**Pull request by veksen**
### Description

The `@hapi/joi` package has moved to `joi`, and the TS definitions are now included with the package.

### Testing

N/A, but I ran the test suite locally.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

Note: I got a warning from using package-lock v1, while the project uses v2. This would explain the huge change of `package-lock.json`. Not sure how to change that, I'm using Node 14 locally.

(by the way, the https://github.com/auth0/open-source-template/blob/main/CONTRIBUTING.md link is a 404)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,7 +104,6 @@
     "typescript": "^4.1.3"
   },
   "dependencies": {
-    "@hapi/joi": "^17.1.1",
```
Comment (adamjmcgrath) [inline package.json:108]: Thanks @veksen - could you add `joi` and make sure the project still compiles? We also still need the types, unless the new version of `joi` comes bundled with them?

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (veksen) [reply package.json:108]: The important part 😅 Added

Comment (veksen) [reply package.json:108]: The types are now part of `joi`, if you want to double check that

**Review by adamjmcgrath — APPROVED** [review]
Thanks @veksen!

**Pull request merged**